src/agents/conflict_detection_agent.py

```
@@ -201,30 +235,61 @@ def detect_conflicts(
-                # Example: Extract character names from chapter text and query for their KB entries.
+        all_kb_chunks_with_scores: Dict[str, float] = {} # Use dict to store unique doc content and its highest score
 
-                kb_results = lore_keeper_instance.kb_manager.query_knowledge_base(novel_id, query_text, n_results=7) # Increased n_results for broader context
+        if lore_keeper_instance and hasattr(lore_keeper_instance, 'kb_manager') and lore_keeper_instance.kb_manager:
+            try:
+                # 1. Targeted Entity Queries
+                extracted_entities = self._extract_key_entities_for_kb_query(current_chapter_text)
+                if extracted_entities:
+                    print(f"ConflictDetectionAgent: Performing targeted KB queries for entities: {extracted_entities}")
+                    for entity in set(extracted_entities): # Use set to avoid duplicate queries for same entity
+                        if not entity: continue
```

Ruff (0.11.9)

Multiple statements on one line (colon)

(E701)